src/main/java/org/greencodeinitiative/creedengo/java/checks/MakeNonReassignedVariablesConstants.java

```diff
@@ -17,20 +17,28 @@ public class MakeNonReassignedVariablesConstants extends IssuableSubscriptionVis
 
     private static final Logger LOGGER = Loggers.get(MakeNonReassignedVariablesConstants.class);
 
+    private final String LOMBOK_SETTER = "Setter";
+    private final String LOMBOK_DATA = "Data";
+
+    private boolean hasParsedImports = false;
+    private boolean hasLombokSetterImport = false;
+    private boolean hasLombokDataImport = false;
+
+
     @Override
     public List<Kind> nodesToVisit() {
         return List.of(Kind.VARIABLE);
```

Comment:
I think you can improve more efficiently the import checks. You can add "Kind.IMPORT" in the list of nodes to check. Because Imports are the first statements analyezed in the code, you can know if lombok importrs are present or not.